src/shared/lib/comparison/ComparisonStore.ts

```diff
@@ -2014,6 +2037,120 @@ export default abstract class ComparisonStore {
+                        studyIds: studyIds,
+                    } as MolecularProfileFilter,
+                });
+            },
+        },
+        []
+    );
+
+    readonly customDriverAnnotationProfiles = remoteData<MolecularProfile[]>(
+        {
+            await: () => [this.molecularProfilesInStudies],
+            invoke: () => {
+                return Promise.resolve(
+                    _.filter(
+                        this.molecularProfilesInStudies.result,
+                        (molecularProfile: MolecularProfile) =>
+                            // discrete CNA's
+                            (molecularProfile.molecularAlterationType ===
+                                AlterationTypeConstants.COPY_NUMBER_ALTERATION &&
+                                molecularProfile.datatype ===
+                                    DataTypeConstants.DISCRETE) ||
+                            // mutations
+                            molecularProfile.molecularAlterationType ===
+                                AlterationTypeConstants.MUTATION_EXTENDED ||
+                            // structural variants
+                            molecularProfile.molecularAlterationType ===
+                                AlterationTypeConstants.STRUCTURAL_VARIANT
+                    )
+                );
+            },
+        },
+        []
+    );
+
+    readonly customDriverAnnotationReport = remoteData<IDriverAnnotationReport>(
+        {
+            await: () => [this.customDriverAnnotationProfiles],
+            invoke: async () => {
+                const molecularProfileIds = _.map(
+                    this.customDriverAnnotationProfiles.result,
+                    (p: MolecularProfile) => p.molecularProfileId
+                );
+                const report = await internalClient.fetchAlterationDriverAnnotationReportUsingPOST(
+                    {
+                        molecularProfileIds,
+                    }
+                );
+                return {
+                    ...report,
+                    hasCustomDriverAnnotations:
+                        report.hasBinary || report.tiers.length > 0,
+                };
+            },
+            onResult: result => {
+                initializeCustomDriverAnnotationSettings(
+                    result!,
+                    this.driverAnnotationSettings,
+                    this.driverAnnotationSettings.customTiersDefault
+                );
+            },
+        }
+    );
+
+    @computed get showDriverAnnotationMenuSection() {
+        return !!(
+            this.customDriverAnnotationReport.isComplete &&
+            this.customDriverAnnotationReport.result!.hasBinary &&
+            AppConfig.serverConfig
+                .oncoprint_custom_driver_annotation_binary_menu_label &&
+            AppConfig.serverConfig
+                .oncoprint_custom_driver_annotation_tiers_menu_label
+        );
+    }
+
+    @computed get showTierAnnotationMenuSection() {
```

Comment:
I think we need better naming.  What kind of tier?  driverTier?

Reply:
ditto for the rest below

Reply:
I renamed all.